### PR TITLE
PR for shovill bam file

### PR DESCRIPTION
**Pull request by pimarin**
Follow the last PR [4514](https://github.com/galaxyproject/tools-iuc/pull/4514) accidentally closed
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [X] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/shovill/shovill.xml [hunk 1]
```diff
@@ -116,6 +118,22 @@
...
+            <discover_datasets pattern="(?P&lt;name&gt;.+\.histogram)" ext="txt" directory="out/" />
+            <discover_datasets pattern="(?P&lt;name&gt;.+\.gz)" ext="gz" directory="out/" />
+            <discover_datasets pattern="(?P&lt;name&gt;.+\.bam)" ext="bam" directory="out/" />
+            <discover_datasets pattern="(?P&lt;name&gt;.+\.amb)" ext="amb" directory="out/" />
+            <discover_datasets pattern="(?P&lt;name&gt;.+\.ann)" ext="ann" directory="out/" />
+            <discover_datasets pattern="(?P&lt;name&gt;.+\.bwt)" ext="bwt" directory="out/" />
+            <discover_datasets pattern="(?P&lt;name&gt;.+\.pac)" ext="pac" directory="out/" />
+            <discover_datasets pattern="(?P&lt;name&gt;.+\.sa)" ext="sa" directory="out/" />
+            <discover_datasets pattern="(?P&lt;name&gt;.+\.uncorrected)" ext="fasta" directory="out/" />
+            <discover_datasets pattern="(?P&lt;name&gt;.+)" directory="out/spades" />
```
Comment (bernt-matthias) [inline tools/shovill/shovill.xml:135]: `. +` should cover everything, or?

File: tools/shovill/shovill.xml [hunk 2]
```diff
@@ -242,6 +260,32 @@
...
+            </output>
+            <output name="shovill_std_log" ftype="txt" >
+                <assert_contents>
+                    <has_text text="Read stats: total_bp = 300000"/>
+                    <has_text text="[lighter] Processed 820 reads"/>
+                    <has_text text="[FLASH] FLASH v1.2.11 complete!"/>
+                    <has_text text="[spades] ======= SPAdes pipeline finished."/>
+                </assert_contents>
+            </output>
+            <output_collection name="keepfile_collection" type="list">
```
Comment (bernt-matthias) [inline tools/shovill/shovill.xml:284]: Can you add the `count` attribute?

File: tools/shovill/shovill.xml (hunk 1 above)
Comment (pimarin) [reply tools/shovill/shovill.xml:135]: This is not good, with `--keepfile` the folder name dependt of the assembler use : if spades --> out/spades but if velvet out/velvet.
Is there a way to discover dataset in a subdirectory with a changing name ?

Comment (bernt-matthias) [reply tools/shovill/shovill.xml:135]: Sure. Check the `recurse` attribute of `discover_datasets`: https://docs.galaxyproject.org/en/master/dev/schema.html#tool-outputs-collection-discover-datasets

Comment (bernt-matthias) [reply tools/shovill/shovill.xml:135]: > `.+` should cover everything, or?

I see.. the separate `discover_datasets` are helpful for setting correct extensions.

File: tools/shovill/shovill.xml [hunk 3]
```diff
@@ -100,6 +101,7 @@
                     </mapping>
                 </sanitizer>
             </param>
+            <param name="keepfiles" argument="--keepfiles" type="boolean" truevalue="--keepfiles" falsevalue=""  label="Keep intermediate files (default: OFF)" help="Keep intermediates files like .bam files (--keepfiles)"/>
```
[the remark this answers is not in the document]
Comment (pimarin) [reply tools/shovill/shovill.xml]: I remove this

File: tools/shovill/shovill.xml (hunk 1 above)
Comment (pimarin) [reply tools/shovill/shovill.xml]: I remove the collection and I just create an option to keep the bam files for user depending of the nocorr option.
I will try next time to implement a proper keefile to have all intermediary files for user if need.

File: tools/shovill/shovill.xml [hunk 4]
```diff
@@ -104,7 +106,13 @@
             <param name="gsize" argument="--gsize" type="text" value="" label="Estimated genome size" help="An estimate of the final genome size, it will autodetect if this is blank. (default: '')" />
             <param name="kmers" argument="--kmers" type="text" value="" label="List of kmer sizes to use" help="List of K-mer sizes to use in SPAdes. Blank is AUTO. default: ''" />
             <param name="opts" argument="--opts" type="text" value="" label="Extra SPAdes options" help="eg. --plasmid --sc ... (default: '')" />
-            <param name="nocorr" argument="--nocorr" type="boolean" truevalue="--nocorr" falsevalue="" checked="True" label="Disable post-assembly correction" help="Disable post assembly correction with pilon (default: ON)" />
+            <conditional name="keep_files">
```
Comment (bgruening) [inline tools/shovill/shovill.xml:109]: conditionals should not use booleans https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#booleans

Can you use a select here?

Comment (pimarin) [reply tools/shovill/shovill.xml:109]: You whrite, I changed it

File: tools/shovill/shovill.xml [hunk 5]
```diff
@@ -104,7 +107,16 @@
...
```
Comment (bernt-matthias) [inline tools/shovill/shovill.xml:110]: Might also be a single boolean with `truevalue="--keepfiles" falsevalue="--nocorr"? `

File: tools/hamronization/hamronize_summarize.xml [hunk 6]
```diff
@@ -17,15 +17,14 @@
...
             --output output.tsv
         #elif $summary_type == "json"
             --output output.json
-        #else 
+        #else
              --output output.html
         #end if
-        *.tsv
-        *.json
+        *.{tsv,json}
```
Comment (bernt-matthias) [inline tools/hamronization/hamronize_summarize.xml:27]: Is this already covered by a test? WOuld be nice.

File: tools/shovill/shovill.xml (hunk 5 above)
Comment (pimarin) [reply tools/shovill/shovill.xml:110]: humm... `--nocorr` is the tool option to remove the post assembly correction and I created an option `keepfile` to filter bam output when post assembly is enable. So to have bam you need to remove `--nocorr` but you can enable post assembly without keep the bam file.

Comment (bernt-matthias) [reply tools/shovill/shovill.xml:110]: Up to you.. I was just guessing around :)

File: tools/hamronization/hamronize_summarize.xml (hunk 6 above)
Comment (bernt-matthias) [reply tools/hamronization/hamronize_summarize.xml:27]: Which one is this? This one: https://github.com/galaxyproject/tools-iuc/pull/4541 ? 

Then we could merge them.

**Comment by pimarin** [comment]
I removed the hamronization tools files for this PR

**Comment by pimarin** [comment]
Hi @gallardoalba,
I will check it, sorry I was in holidays, I'll work on the galaxy tool during the end or maybe next week !

**Comment by pimarin** [comment]
commits seem to be lock due to my mistake with tools/hamronization/hamronize_summarize.xml file which I tried to remove from commits

**Pull request closed without merging**